packages/kbn-config-schema/src/types/array_type.ts

```diff
@@ -49,6 +49,8 @@ export class ArrayType<T> extends Type<T[]> {
       case 'any.required':
       case 'array.base':
         return `expected value of type [array] but got [${typeDetect(value)}]`;
+      case 'array.sparse':
+        return `sparse array are not allowed`;
```

Comment:
I didn't add JSON.stringify because 
```js
JSON.stringify([1, undefined]) === '[1,null]'
```
which is even more confusing

Reply:
I think it's fine. That would also results in enormous error messages when working with array of objects.